src/Relation/Binary/Reasoning/Syntax.agda

```diff
@@ -0,0 +1,366 @@
+-- "natural" order and syntax declarations are later used to re-order
+-- them. This is because the `step` ordering allows the type-checker to
+-- better infer the middle argument `y` from the `_IsRelatedTo_`
+-- argument (see issue 622).
+--
+-- This has two practical benefits. First it speeds up type-checking by
+-- approximately a factor of 5. Secondly it allows the combinators to be
+-- used with macros that use reflection, e.g. `Tactic.RingSolver`, where
+-- they need to be able to extract `y` using reflection.
+
+------------------------------------------------------------------------
+-- Syntax for unidirectional relations
+
+-- See https://github.com/agda/agda-stdlib/issues/2150 for a possible
+-- simplification.
+
+module _
+  {R : REL A B ℓ₂}
+  (S : REL B C ℓ₁)
+  (T : REL A C ℓ₃)
+  (step : Trans R S T)
+  where
+
+  forward : ∀ (x : A) {y z} → S y z → R x y → T x z
+  forward x yRz x∼y = step {x} x∼y yRz
+
+
+  -- Preorder syntax
+  module ∼-syntax where
+    infixr 2 step-∼
+    step-∼ = forward
+    syntax step-∼ x yRz x∼y = x ∼⟨ x∼y ⟩ yRz
+
+
+  -- Partial order syntax
+  module ≤-syntax where
+    infixr 2 step-≤
+    step-≤ = forward
+    syntax step-≤ x yRz x≤y = x ≤⟨ x≤y ⟩ yRz
+
+
+  -- Strict partial order syntax
+  module <-syntax where
+    infixr 2 step-<
+    step-< = forward
+    syntax step-< x yRz x<y = x <⟨ x<y ⟩ yRz
+
+
+  -- Subset order syntax
+  module ⊆-syntax where
+    infixr 2 step-⊆
+    step-⊆ = forward
+    syntax step-⊆ x yRz x⊆y = x ⊆⟨ x⊆y ⟩ yRz
+
+
+  -- Strict subset order syntax
+  module ⊂-syntax where
+    infixr 2 step-⊂
+    step-⊂ = forward
+    syntax step-⊂ x yRz x⊂y = x ⊂⟨ x⊂y ⟩ yRz
+
+
+  -- Square subset order syntax
+  module ⊑-syntax where
+    infixr 2 step-⊑
+    step-⊑ = forward
+    syntax step-⊑ x yRz x⊑y = x ⊑⟨ x⊑y ⟩ yRz
+
+
+  -- Strict square subset order syntax
+  module ⊏-syntax where
+    infixr 2 step-⊏
+    step-⊏ = forward
+    syntax step-⊏ x yRz x⊏y = x ⊏⟨ x⊏y ⟩ yRz
+
+
+  -- Divisibility syntax
+  module ∣-syntax where
+    infixr 2 step-∣
+    step-∣ = forward
+    syntax step-∣ x yRz x∣y = x ∣⟨ x∣y ⟩ yRz
+
+
+  -- Single-step syntax
+  module ⟶-syntax where
+    infixr 2 step-⟶
+    step-⟶ = forward
+    syntax step-⟶ x yRz x∣y = x ⟶⟨ x∣y ⟩ yRz
+
+
+  -- Multi-step syntax
+  module ⟶*-syntax where
+    infixr 2 step-⟶*
+    step-⟶* = forward
+    syntax step-⟶* x yRz x∣y = x ⟶*⟨ x∣y ⟩ yRz
+
+
+------------------------------------------------------------------------
+-- Syntax for bidirectional relations
+
+  module _
+    {U : REL B A ℓ₄}
+    (sym : Sym U R)
+    where
+
+    backward : ∀ x {y z} → S y z → U y x → T x z
+    backward x yRz x≈y = forward x yRz (sym x≈y)
+
+    -- Setoid equality syntax
+    module ≈-syntax where
+      infixr 2 step-≈ step-≈˘
+      step-≈ = forward
+      step-≈˘ = backward
+      syntax step-≈  x yRz x≈y = x ≈⟨ x≈y ⟩ yRz
+      syntax step-≈˘ x yRz y≈x = x ≈˘⟨ y≈x ⟩ yRz
```

Comment:
Coming, as ever, late to this: in #2268 I've had to use explicit `x ≈⟨ refl ⟩...` steps, because there is no abbreviation `≈⟨⟩` declared, whereas `≡⟨⟩` *is* available as an alternative. Was this what you intended, or should there be a bug-fixing issue/PR raised to add the additional syntax?

Reply:
Yes, the intent is that people should use `≡⟨⟩` not `≈⟨⟩`